### PR TITLE
feat: issue-439/calendar-modal

### PR DESCRIPTION
**Pull request by Bct-crypto**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,6 +45,7 @@
     "axios": "^1.6.0",
     "class-variance-authority": "^0.7.0",
     "clsx": "^2.0.0",
+    "flowbite": "^2.5.2",
```
Comment (dharamveergit) [inline package.json:48]: Please use [Headless UI's Dialog](https://headlessui.com/react/dialog) to create a modal without installing additional libraries.

File: src/components/development-pages/calendar-modal.astro [hunk 2]
```diff
@@ -0,0 +1,78 @@
+---
+import Calenderbutton from "./calenderbutton.astro";
+import TimeZoneCalendar from "./time-zone-calendar";
+import { CalendarIcon } from "@heroicons/react/20/solid";
+---
+
+<div>
+  <!-- Modal toggle -->
```
Comment (dharamveergit) [inline src/components/development-pages/calendar-modal.astro:8]: Don't add comments

File: src/layouts/layout.astro [hunk 3]
```diff
@@ -46,7 +46,7 @@ const pathname = Astro.url.pathname;
 
   <Header />
 
-  <main>
+  <main class="relative">
```
Comment (dharamveergit) [inline src/layouts/layout.astro:49]: Using relative to the entire website layout might have some side effects.

File: tailwind.config.cjs [hunk 4]
```diff
@@ -123,5 +124,5 @@ module.exports = {
       },
     },
   },
-  plugins: [require("@tailwindcss/typography"), require("tailwindcss-animate")],
+  plugins: [require("@tailwindcss/typography"), require("tailwindcss-animate"), require('flowbite/plugin')],
```
Comment (dharamveergit) [inline tailwind.config.cjs:127]: No need to install another library; you can use Headless UI for the modal.

File: src/components/development-pages/calendar-modal.astro [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+      <!-- Modal content -->
+      <div class="relative rounded-lg bg-background2 py-8 shadow">
+        <!-- Modal header -->
+        <div
+          class="flex items-center justify-between rounded-t px-4 md:px-8 xl:px-12"
+        >
+          <h3 class="text-xl font-semibold text-gray-900 dark:text-white">
+            Akash Community Groups Calendar
+          </h3>
+          <button
```
Comment (dharamveergit) [inline src/components/development-pages/calendar-modal.astro:37]: Try using the global button component.

File: src/components/development-pages/calendar-modal.astro [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    <div
+      class="relative max-h-full w-full max-w-screen-2xl px-4 sm:px-8 md:px-12 lg:px-16 xl:px-20"
+    >
+      <!-- Modal content -->
+      <div class="relative rounded-lg bg-background2 py-8 shadow">
+        <!-- Modal header -->
+        <div
+          class="flex items-center justify-between rounded-t px-4 md:px-8 xl:px-12"
+        >
+          <h3 class="text-xl font-semibold text-gray-900 dark:text-white">
```
Comment (dharamveergit) [inline src/components/development-pages/calendar-modal.astro:34]: No need to add colors like white or black to the text, as it's handled globally.

**Review by dharamveergit — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (Bct-crypto) [reply package.json:48]: can I use https://www.radix-ui.com/primitives/docs/components/dialog

File: src/components/development-pages/calendar-modal.astro (hunk 2 above)
Comment (Bct-crypto) [reply src/components/development-pages/calendar-modal.astro:8]: got it

File: tailwind.config.cjs (hunk 4 above)
Comment (Bct-crypto) [reply tailwind.config.cjs:127]: you are using Headless UI 1.7
but the latest version is 2.2
I think flowbite is better than headlessui.
https://npmtrends.com/flowbite-vs-headlessui-vs-tailwind-elements-vs-tailwindcss

Comment (dharamveergit) [reply tailwind.config.cjs:127]: But we are using Headless UI everywhere, so use that. The version will not affect any required functionality—it's just a modal, nothing major.

File: package.json (hunk 1 above)
Comment (dharamveergit) [reply package.json:48]: Why is it necessary to install another library (I know other component from radix is there but dialog is not) when Headless UI can be used?

**Comment by HoomanDgtl** [comment]
this pr is addressed in #451 , thanks @Bct-crypto for your contributions

**Comment by Bct-crypto** [comment]
welcome

**Pull request closed without merging**